nkms/storage/encrypted_file.py

```diff
@@ -120,51 +47,28 @@ def _read_chunk(self, chunk_size, nonce):
         :return: Decrypted/Authenticated chunk
         :rtype: Bytes
         """
-        ciphertext = self.file_obj.read(chunk_size)
+        ciphertext = self.file_obj.read(chunk_size + PADDING_LENGTH)
         return self.cipher.decrypt(ciphertext, nonce=nonce)
 
-    def open_new(self, keys, chunk_size=1000000, nonce=None):
-        """
-        Opens a new EncryptedFile and creates a header for it ready for
-        writing encrypted data.
-
-        :param list keys: Encrypted keys to put in the header.
-        :param int chunk_size: Size of encrypted chunks in bytes, default is 1MB
-        :param bytes nonce: 20 byte Nonce to use for encryption
-        """
-        self.file_obj = open(self.path, mode=self.mode)
-        self._build_header(nonce=nonce, keys=keys, chunk_size=chunk_size)
-
-    def open(self, is_new=False):
-        """
-        Opens a file for Encryption/Decryption.
-
-        :param bool is_new: Is the file new (and empty)?
-        """
-        # TODO: Error if self.file_obj is already defined
-        self.file_obj = open(self.path, mode=self.mode)
-
-        # file_obj is now ready for reading/writing encrypted data
-        if not is_new:
-            self._read_header()
-
     def read(self, num_chunks=0):
         """
         Reads num_chunks of encrypted ciphertext and decrypt/authenticate it.
 
-        :param int num_chunks: Number of chunks to read. Default is all chunks
+        :param int num_chunks: Number of chunks to read. When set to 0, it will
+            read the all the chunks and decrypt them.
 
         :return: List of decrypted/authenticated ciphertext chunks
         :rtype: List
         """
-        if num_chunks == 0:
-            num_chunks = self.header['chunks']
+        if not num_chunks:
+            num_chunks = self.header[b'num_chunks']
```

Comment:
After unpacking with `msgpack`, the keys will always be `bytes`. So, rather than deal with that logic, I just made the keys always be of type `bytes`.

Reply:
Yeah, it's weird, but if you do `msgpack.loads(msgpack.dumps({'a': 1}))`, you'll get `{b'a': 1}`. That's why :-)

Reply:
I think it's a little odd to have a public interface that's a dict where the keys are bytes.  But that's more the discussion in my comment above - do we really want this to be a public interface?